src/bootloaders/bootloader.h

```diff
@@ -38,7 +38,9 @@ typedef enum {
         BOOTLOADER_CAP_UEFI = 1 << 1,   /**<Bootloader supports UEFI */
         BOOTLOADER_CAP_GPT = 1 << 2,    /**<Bootloader supports GPT boot partition */
         BOOTLOADER_CAP_LEGACY = 1 << 3, /**<Bootloader supports legacy boot */
-        BOOTLOADER_CAP_MAX = 1 << 4
```

Comment:
So originally we were trying to distinguish an EFI booting with gpt vs extlinux using gpt vs grub on mbr iirc. I am unsure how correct that statement is now. I think we should reconsider this design because it looks a bit awkward currently. Nothing to change now though.

Reply:
The big problem is that we should reuse the existing bootloader instead of bluntly assuming grub even exists on the system. This broke one of my KVM systems that has to boot with MBR support, since no grub in clearlinux but syslinux we have.

IOW, a big part of the right solution is checking if the appropriate binaries exist, since then we can have overlapping features and get away with it.

Reply:
The reuse case makes sense but I think this gets into a design change that is needed for cbm to do the right thing in more cases (just install the kernel when it isn't in charge of the bootloader, handle both syslinux and systemd-boot both being feasable options, etc). The file existence check is a nice short cut for now but leaves us open to future problems if all things exist.

Reply:
I see your point, I collecting a list of things I see that need change designwise, this is one of'em.